pkg/kube/kube_handler.go

```diff
@@ -726,6 +727,266 @@ func doValidationRequest(cli httpclient.Client, req *http.Request) (*ValidationR
+	return &mediaType, nil
+}
+
+//compareMediaType matches wantStr with the inputStr
+func compareMediaType(repoName string, inputStr string, wantStr string) (bool, error) {
+
+	err := fmt.Errorf("%v is not a Helm OCI Repo", repoName)
+	result := false
+
+	tmpStr1 := regexp.MustCompile(`[\\/\\+]`).Split(inputStr, -1)
+	if len(tmpStr1) < 2 {
+		return result, err
+	}
+
+	tmpStr2 := regexp.MustCompile(`.[^.]*$`).Split(tmpStr1[1], -1)
+	if len(tmpStr2) == 0 {
+		return result, err
+	}
+	gotStr := tmpStr2[0]
+
+	log.Infof("gotStr : (%v)\n", gotStr)
+	if gotStr != wantStr {
+		return result, err
+	}
+	return true, nil
+}
+
+// ValidateOCIAppRepository  validates OCI Repos only
+func ValidateOCIAppRepository(appRepo *v1alpha1.AppRepository, cli httpclient.Client) (bool, error) {
+
+	var err error
+
+	result := false
+	err = nil
+
+	repoURL := strings.TrimSuffix(strings.TrimSpace(appRepo.Spec.URL), "/")
+
+	// For the OCI case, we want to validate that all the given repositories are valid
+	if len(appRepo.Spec.OCIRepositories) == 0 {
+		return false, ErrEmptyOCIRegistry
+	}
+	for _, repoName := range appRepo.Spec.OCIRepositories {
+		log.Infof("repoURL : %v, repoName : %v\n", repoURL, repoName)
+		tagVersionPtr, err := getOCIAppRepositoryTag(cli, repoURL, repoName)
+		if err != nil {
+			return false, err
+		}
+
+		log.Infof("tagVersion: %v", *tagVersionPtr)
+
+		mediaTypePtr, err := getOCIAppRepositoryMediaType(cli, repoURL, repoName, *tagVersionPtr)
+		if err != nil {
+			return false, err
+		}
+
+		log.Infof("mediaType : %v", *mediaTypePtr)
+
+		wantStr := "vnd.cncf.helm.config"
+
+		result, err = compareMediaType(repoName, *mediaTypePtr, wantStr)
+		if !result {
+			return result, err
+		}
+	}
+	return result, err
+}
+
+type nonOCIRepoRequests struct {
+	reqs []*http.Request
+}
+
+type OCIRepoRequests struct {
+	appRepo *v1alpha1.AppRepository
+}
+
+type HttpValidator interface {
+	IsValid(cli httpclient.Client) (*ValidationResponse, error)
+}
+
+func (nonOCIRepo_reqs nonOCIRepoRequests) IsValid(cli httpclient.Client) (*ValidationResponse, error) {
+	reqs := nonOCIRepo_reqs.reqs
+	var err = error(nil)
+
+	response := &ValidationResponse{}
+
+	for _, req := range reqs {
+		res, err := cli.Do(req)
+		if err != nil {
+			// If the request fail, it's not an internal error
+			return &ValidationResponse{Code: 400, Message: err.Error()}, nil
+		}
+		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to parse validation response. Got: %v", err)
+		}
+		response = &ValidationResponse{Code: res.StatusCode, Message: string(body)}
+		if response.Code != 200 {
+			return response, nil
+		}
+	}
+
+	return response, err
+}
+
+func (OCIRepo_reqs OCIRepoRequests) IsValid(cli httpclient.Client) (*ValidationResponse, error) {
+
+	var response *ValidationResponse
+	response = &ValidationResponse{Code: 200, Message: ""}
+
+	result, err := ValidateOCIAppRepository(OCIRepo_reqs.appRepo, cli)
+	if !result {
```

Comment:
Things like this read much more nicely if you chose a var name like `isValid` rather than `result`. That way, it can read as "if it isn't valid then"